### PR TITLE
Update to support Apache 2.4 - rework of #136

### PR DESCRIPTION
**Pull request by wyardley**
This reworks #136  by @steveames
Still probably could use some review, though it does get the tests passing and is rebased against current master.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -6,21 +6,23 @@
...
 
-  # puppetlabs/apache names the apache service
-  # resource 'httpd', nevermind the actual os
-  $apache_service = 'httpd'
-
   case $::osfamily {
     'Debian': {
-      if $::operatingsystem == ubuntu {
-        $apache_confd = '/etc/apache2/conf-enabled'
+      if ($::operatingsystem == 'ubuntu') {
```
Comment (juniorsysadmin) [inline manifests/params.pp]: You probably should use upper case naming to be consistent. Unless `$::osfamily` cares about case and `$::operatingsystem` Puppet logic does not

Comment (bastelfreak) [reply manifests/params.pp]: can you please switch to the new facts hash? $facts['os']['family']

File: manifests/params.pp [hunk 2]
```diff
@@ -6,21 +6,23 @@
 #
 class puppetboard::params {
 
-  # puppetlabs/apache names the apache service
-  # resource 'httpd', nevermind the actual os
-  $apache_service = 'httpd'
-
   case $::osfamily {
```
Comment (bastelfreak) [inline manifests/params.pp]: please use the facts hash here

File: manifests/params.pp [hunk 3]
```diff
@@ -6,21 +6,23 @@
...
     'Debian': {
-      if $::operatingsystem == ubuntu {
-        $apache_confd = '/etc/apache2/conf-enabled'
+      if ($::operatingsystem == 'ubuntu') {
+        if (versioncmp($::operatingsystemrelease,'14.04')) {
```
Comment (bastelfreak) [inline manifests/params.pp]: please use the facts hash here. Probably $facts['os']['release']['full']

File: manifests/params.pp (hunk 2 above)
Comment (wyardley) [reply manifests/params.pp]: I did see those, and had left them the same as the original PR. I added an additional commit that should switch to structured facts. 

Acceptance tests are another story; they were out of date, so I'm working on another pull to at least somewhat fix them.

**Pull request merged**